doc/Core.xml

```diff
@@ -6229,6 +6229,42 @@ onvif://www.onvif.org/name/ARV-453
           </variablelist>
         </para>
       </section>
+      <section>
+        <title>Configuration Renewal</title>
+        <para>
+          The configuration allows for a renewal endpoint to be set. If the device supports this feature, it shall automatically renew the credentials 
+          when they are about to expire.
+        </para>
+        <para>
+          The device shall do a GET request to the configured <literal>RenewalEndpoint</literal> with a JWT token retrieved from the configured 
+          <literal>AuthorizationServer</literal>. The endpoint shall respond with a JSON payload with the following structure:
+          <programlisting><![CDATA[{
+    "region": "string" | null,
+    "storageUri": "string" | null,
+    "user": {
+        "username": "string" | null,
+        "password": "string" | null,
+        "token": "string" | null
+    },
+    "expiresAt": "<ISO 8601 date-time>"
+}]]></programlisting>
+          Any null value in the response shall clear the corresponding optional parameter in the <literal>StorageConfiguration</literal>.
+        </para>
+        <para>
+          When the device receive a configuration with the <literal>ConfigurationRenewal</literal> set, it shall immediately contact
+          the renewal endpoint to get up-to-date credentials. The device may use the credentials provided by the configuration in the
+          meantime to avoid service disruption.
+        </para>
+        <para>
+          Once the device has called the renewal endpoint, the corresponding <literal>StorageConfiguration</literal> shall be updated with the
```

Comment:
Is it really necessary to update the static configuration on flash? Besides the unwanted side affect of writing to flash unnecessarily often it does not seem logical to change the static configuration with the dynamic fetching of tokens.

Reply:
Since the alternative was pushing an update to the configuration anyway it doesn't feel too different. Of course the requirement does not mandate the device writes it to persistent storage, as long as it gives the right response on GetStorageConfiguration the device may remember it anywhere it wants I feel.

The main reason I see this as needed is to facilitate troubleshooting in case something goes wrong. The client can query the device state to figure out what is currently in use and attempt to resolve the situation